[extension][memo] add new extension: memo

mt40 opened the pull request:
## Description

Memo is an extension for saving copied content to Notion.
This extension provides 2 commands:
1. Save copied text in Notion
2. Save copied text in most recent page you worked on. 

Vision is to support more copied content like images, and more output like Apple Notes, OneNote, ...

## Screencast

<img width="1000" alt="memo-0" src="https://user-images.githubusercontent.com/8610576/208226857-b3c577a3-56d0-49ad-8cee-0b2c87061e17.png">
<img width="1000" alt="memo-1" src="https://user-images.githubusercontent.com/8610576/208226885-806c6bfc-58a8-41df-98a6-b12726b95d49.png">
<img width="1000" alt="memo-2" src="https://user-images.githubusercontent.com/8610576/208226890-2acebe2e-7c3b-4a7b-bcda-1aa83e33a1c7.png">



## Checklist

- [x] I read the [extension guidelines](https://developers.raycast.com/basics/prepare-an-extension-for-store)
- [x] I read the [documentation about publishing](https://developers.raycast.com/basics/publish-an-extension)
- [x] I ran `npm run build` and [tested this distribution build in Raycast](https://developers.raycast.com/basics/prepare-an-extension-for-store#metadata-and-configuration)
- [x] I checked that files in the `assets` folder are used by the extension itself
- [x] I checked that assets used by the `README` are placed outside of the `metadata` folder

pernielsentikaer commented on extensions/memo/src/components/SaveToLast.tsx:
```diff
@@ -0,0 +1,90 @@
+            <Form.TextArea id="text" title={"Copied text"} value={props.copiedText} onChange={setText} />
+
+            {lastPageSection}
+
+            <Form.Separator />
+            <Form.Description
+                text="If you like this extension, consider buying us a coffee by clicking <Actions> below.
+Thank you 🤝"
+            />
```
```suggestion
```
The Action in the [ActionPanel](https://developers.raycast.com/api-reference/user-interface/actions#api-reference) is okay; this will be a bit too much

mt40 replied:
sure, I removed this

pernielsentikaer commented on extensions/memo/src/components/SaveToLast.tsx:
```diff
@@ -0,0 +1,90 @@
+            </ActionPanel.Section>
+
+            <ActionPanel.Section>
+                <Action.OpenInBrowser url={DONATION_URL} title={"Buy us a coffee"} icon="bmcbrand/bmc-logo.svg" />
```
```suggestion
                <Action.OpenInBrowser url={DONATION_URL} title="Buy Us a Coffee" icon="bmcbrand/bmc-logo.svg" />
```

pernielsentikaer commented on extensions/memo/src/components/SaveToPage.tsx:
```diff
@@ -0,0 +1,95 @@
+            </ActionPanel.Section>
+
+            <ActionPanel.Section>
+                <Action.OpenInBrowser url={DONATION_URL} title={"Buy us a coffee"} icon="bmcbrand/bmc-logo.svg" />
```
```suggestion
                <Action.OpenInBrowser url={DONATION_URL} title="Buy Us a Coffee" icon="bmcbrand/bmc-logo.svg" />
```

pernielsentikaer commented on extensions/memo/src/components/SaveToPage.tsx:
```diff
@@ -0,0 +1,95 @@
+                    id={"url_title"}
+                />
+            )}
+
+            <Form.Separator />
+
+            <Form.Description
+                text="If you like this extension, consider buying us a coffee by clicking <Actions> below.
+Thank you 🤝"
+            />
```
```suggestion
```
The Action in the [ActionPanel](https://developers.raycast.com/api-reference/user-interface/actions#api-reference) is okay; this will be a bit too much

pernielsentikaer reviewed (changes requested):
Hi 👋 

Thanks for your contribution 💪 

I have now tested your extension, and I have some feedback ready for you:

* It seems like there is some flicker when opening the command the first time

https://user-images.githubusercontent.com/38033107/208847427-f04a5ae7-c733-464a-9041-1ff440981ac0.mp4

* Some weird flicker going on while typing

https://user-images.githubusercontent.com/38033107/208848346-5f8e7e69-6a43-41f7-88cd-f678a7a6a9cb.mp4

* In this view, it's not possible to modify the content at all; that seems like a bug

https://user-images.githubusercontent.com/38033107/208848520-6d7c5fd0-d00f-404b-afad-533a87124c0c.mp4

We might also have to host the proxy, let's look into that a bit later 😊




I'm looking forward to testing this extension again 🔥 

Request a new review when you are ready. Feel free to contact me here or at [Slack](https://www.raycast.com/community) if you have any questions.

mt40 commented:
> Hi 👋
> 
> Thanks for your contribution 💪
> 
> I have now tested your extension, and I have some feedback ready for you:
> 
> * It seems like there is some flicker when opening the command the first time
> 
>  Raycast-l4RgvRL5.1.mp4 
> 
> * Some weird flicker going on while typing
> 
>  Raycast-U9p5kfSa.mp4 
> 
> * In this view, it's not possible to modify the content at all; that seems like a bug
> 
>  Raycast-cq9VRQCK.mp4 
> 
> We might also have to host the proxy, let's look into that a bit later 😊
> 
> I'm looking forward to testing this extension again 🔥
> 
> Request a new review when you are ready. Feel free to contact me here or at [Slack](https://www.raycast.com/community) if you have any questions.

Hi @pernielsentikaer I have fixed the flickers, which are caused by loading states.
The text can be edited now too inside command `Save to Last Notion Page`.

About the proxy, I host it here: https://oauth-proxy.memo.pluqin.io/

pernielsentikaer reviewed (changes requested):
Found a few more @mt40

Could you look into this and then we'll do a full review after 😊

I'm looking forward to testing this extension again 🔥 

Request a new review when you are ready. Feel free to contact me here or at [Slack](https://www.raycast.com/community) if you have any questions.

mt40 commented on extensions/memo/package.json:
```diff
@@ -0,0 +1,67 @@
+{
+    "$schema": "https://www.raycast.com/schemas/extension.json",
+    "name": "memo",
+    "title": "Memo",
```
This one I think we should not change, since my goal is to support other apps as well, not only Notion

pernielsentikaer reviewed (changes requested):
Hi 👋 

Thanks for your contribution 💪 

I have now tested your extension, and I have some feedback ready for you:

* When entering the commands, I see a toast without any text

https://user-images.githubusercontent.com/38033107/210723181-dffe35a0-42c9-477c-9a46-ee45d300a19f.mp4

* When writing it shows a toast with parsing on every keypress

https://user-images.githubusercontent.com/38033107/210723499-bcd4f82b-8f74-4a08-832c-36c8db946a68.mp4

* It Could be nice with the Notion icon here too

![Raycast-LjQinQGC@2x](https://user-images.githubusercontent.com/38033107/210723791-637d1eb5-2f93-4322-ba8b-f773e3fbc20f.png)

I'm looking forward to testing this extension again 🔥 

Request a new review when you are ready. Feel free to contact me here or at [Slack](https://www.raycast.com/community) if you have any questions.

mt40 commented:
> 
Hi @pernielsentikaer 

1. Those empty toasts signal the success of my operations. I have added messages to them to clarify.
2. This is because I'm trying to parse the text to detect URL. Looking at it again, I think it is quite annoying to show at every key stroke so I remove it.
3. Added the Notion icon for more beauty <3

pernielsentikaer reviewed (approved):
Hi 👋 

Looks good to me, approved 🔥
